ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/helpcenter/searcharticlepage/SearchArticlePresenterTest.java

```diff
@@ -0,0 +1,297 @@
+        searchArticleList = Collections.singletonList(searchArticle);
+    }
+
+    @Test
+    public void initPage() throws Exception {
+        // Act
+        searchArticlePresenter.initPage();
+
+        // Assert
+        verify(mView).showBlankView();
+    }
+
+    @Test
+    public void searchArticles() throws Exception {
+        // Act
+        searchArticlePresenter.searchArticles(TEST_VALUE);
+
+        String returnedQuery = (String) Whitebox.getInternalState
+                (searchArticlePresenter, "mQuery");
+
+        // Assert
+        verify(mView).startSearchTask();
+        assertThat(returnedQuery, is(returnedQuery));
+    }
+
+    @Test
+    public void searchArticlesWithNullQuery() throws Exception {
+        // Act
+        searchArticlePresenter.searchArticles("");
+
+        // Assert
+        verify(mView).cancelBackgroundTasks();
+        verify(mView).showBlankView();
+    }
+
+    @Test
+    public void clearSearchResults() throws Exception {
+        // Act
+        searchArticlePresenter.clearSearchResults();
+
+        // Assert
+        verify(mView).showBlankView();
+    }
+
+    @Test
+    public void loadDataInBackground() throws Exception {
+        // Arrange
+        Whitebox.setInternalState(searchArticlePresenter, "mQuery", TEST_VALUE);
+
+        when(mData.searchArticles(TEST_VALUE, 0, 20)).thenReturn(Collections
+                .singletonList(searchArticle));
+
+        // Act
+        boolean returned = searchArticlePresenter.loadDataInBackground();
+
+        // Assert
+        verify(mData).searchArticles(TEST_VALUE, 0, 20);
+        assertTrue(returned);
+    }
+
+    @Test
+    public void onDataLoadedLessThanRequestLimit() throws Exception {
+        // Arrange
+        List<BaseListItem> mListItems = convertToBaseList(searchArticleList);
+        Whitebox.setInternalState(searchArticlePresenter, "mListItems",
+                mListItems);
+
+        // Act
+        searchArticlePresenter.onDataLoaded(true);
+
+        // Assert
+        verify(mView).showOnlyListView();
+        verify(mView).setUpList(mListItems);
+        verify(mView).setListHasMoreItems(false);
+    }
+
+    @Test
+    public void onDataLoadedGreaterThanRequestLimit() throws Exception {
+        // Arrange
+        List<SearchArticle> searchArticleListManyItems = new ArrayList<>();
+        for (int i = 0; i < 22; i++) {
+            searchArticleListManyItems.add(searchArticle);
+        }
+
+        List<BaseListItem> baseListItems = convertToBaseList
+                (searchArticleListManyItems);
+
+        Whitebox.setInternalState(searchArticlePresenter, "mListItems",
+                baseListItems);
+
+        // Act
+        searchArticlePresenter.onDataLoaded(true);
+
+        // Assert
+        verify(mView).setUpList(baseListItems);
+        verify(mView).showOnlyListView();
+        verify(mView).setListHasMoreItems(true);
+    }
+
+    @Test
+    public void onDataLoadedEmoty() throws Exception {
```

Comment:
⚠️  This snippet of code contains typos (`Emoty`). Spelling errors make reading and understanding the source code harder - mild errors cause the reader to stumble for a fraction of a second, many of them can cause considerable delays; really bad errors can render source code completely unreadable.  

**[More details](https://confluence.devfactory.com/x/QAPvEw)**
          
## The possible solution in this case

```JAVA
public void onDataLoadedEmpty() throws Exception { 

```
 

<a id="5eb9e440-52d6-11e8-9c23-69d6096f3efc" href="none" codereview-uuid-reserved-tag />